src/gluonts/nursery/autogluon_tabular.py

```diff
@@ -103,34 +125,42 @@ def to_forecast(
 
 
 class TabularEstimator(Estimator):
-    def __init__(self, freq: str, prediction_length: int, **kwargs) -> None:
+    def __init__(self, freq: str, prediction_length: int, use_lag=False, lags=[], **kwargs) -> None:
         super().__init__()
         self.task = task
         self.freq = freq
         self.prediction_length = prediction_length
         default_kwargs = {
+            "eval_metric": "mean_absolute_error",
             "excluded_model_types": ["KNN", "XT", "RF"],
             "presets": [
-                "high_quality_fast_inference_only_refit",
-                "optimize_for_deployment",
+                "high_quality_fast_inference_only_refit", "optimize_for_deployment"
             ],
-            "eval_metric": "mean_absolute_error",
         }
         self.kwargs = {**default_kwargs, **kwargs}
+        self.use_lag = use_lag
+        if not self.use_lag:
+            self.lags = []
+        else:
+            self.lags = sorted(list(set(lags))) if lags else list(get_lags_for_frequency(self.freq))
 
     def train(self, training_data: Dataset) -> TabularPredictor:
+        # every time there is only one time series passed
+        # list(training_data)[0] is essentially getting the only time series
         dfs = [
-            get_prediction_dataframe(to_pandas(entry))
+            get_prediction_dataframe(series=to_pandas(entry),
+                                     prediction_length=self.prediction_length,
+                                     use_lag=self.use_lag,
+                                     context_length=len(self.lags), train=True)
             for entry in training_data
         ]
         df = pd.concat(dfs)
-
-        ag_model = self.task.fit(
-            df, label="target", problem_type="regression", **self.kwargs
-        )
-
-        return TabularPredictor(ag_model, self.freq, self.prediction_length)
+        ag_model = self.task.fit(df, label="target", problem_type="regression", output_directory="Eval2/electricity30",
```

Comment:
Is the output directory supposed to be hard-coded here?

Reply:
Sorry, it should be set by users. I'll put it into the kwargs instead.